Don't reject feed blocked group if it has enough active documents

hmusum opened the pull request:
Prefer a feed blocked group before one without coverage (i.e. one that is in maintenance).

https://github.com/vespa-engine/vespa/issues/33755

hmusum committed:
Don't reject feed blocked group if it is the only one with coverage

Prefer a feed blocked group before one without coverage (i.e.
one that is in maintenance).

bratseth commented:
There's always another corner case inside the corner case with this :-)

Would it be easier to understand this if we instead made the condition for rejecting a group 
(is blocking feed && has fewer docs than the max observed)?

hmusum commented:
Yes, makes sense, I'll look into it

hmusum committed:
Reject groups that are feed blocked and have fewer documents than max documents in any group

hmusum committed:
Fix test

(we now select a group that is not feed blocked if possible
instead of the first one that is feed blocked)

hmusum commented:
@bratseth please take a look
